lib/services/local/generic/generic_wrapper.go

```diff
@@ -62,6 +63,24 @@ type ServiceWrapper[T types.ResourceMetadata] struct {
 	service *Service[resourceMetadataAdapter[T]]
 }
 
+// WithPrefix will return a service wrapper with the given parts appended to the backend prefix.
+func (s ServiceWrapper[T]) WithPrefix(parts ...string) *ServiceWrapper[T] {
```

Comment:
It feels like it'd be good to have a test for this method as it's load-bearing for the notifications service. ;)